### PR TITLE
Feature/staking page

### PR DESCRIPTION
**Pull request by worrex**
Added staking page with placeholder values for price, total amount staked and several other values

Fixed some typos

## Related Issues

---

## Checklist:

- [x] I have read [Migaloo's contribution guidelines](https://github.com/White-Whale-Defi-Platform/migaloo-frontend/blob/main/CONTRIBUTING.md).
- [x] My pull request has a sound title and description (not something vague like `Update index.md`)
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation.
- [x] The code is formatted properly `yarn lint`.
- [x] The project builds and is able to deploy on Netlify `yarn build`.

### CODE REVIEW

**Comment by Sen-Com** [comment]
Hit the linked issue limit. #192 is also relevant

**Comment by Sen-Com** [comment]
Added #195

File: components/Pages/Swap/hooks/useTransaction.tsx [hunk 1]
```diff
@@ -56,19 +56,19 @@ type Params = {
...
-  senderAddress,
-  msgs,
-  encodedMsgs,
-  amount,
-  price,
-  onBroadcasting,
-  onSuccess,
-  onError,
-}: Params) => {
+                                 enabled,
```
Comment (0xFable) [inline components/Pages/Swap/hooks/useTransaction.tsx:59]: This formatting looks a little off, do you have like 8 tabs as your setting ? Maybe fine

File: components/Pages/Dashboard/hooks/useUnbonding.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+  );
+  const unbondingPeriod = 60 * 1_000_000_000
+
+  const currentTimeInNanoseconds = Date.now() * 1_000_000;
+
+  const filteredAmpWhaleUnbondingRequests = unbondingInfos?.[0]?.unbonding_requests.filter(req=>((Number(req.timestamp)+unbondingPeriod) > currentTimeInNanoseconds))
+  const filteredBWhaleUnbondingRequests = unbondingInfos?.[1]?.unbonding_requests.filter(req=>((Number(req.timestamp)+unbondingPeriod) > currentTimeInNanoseconds))
+  const filteredUnbondingRequests: UnbondingRequest[] = [...(filteredAmpWhaleUnbondingRequests != null ? filteredAmpWhaleUnbondingRequests : [] ), ...(filteredBWhaleUnbondingRequests != null ? filteredBWhaleUnbondingRequests : [] )]
+
+  const unbondingAmpWhale = convertMicroDenomToDenom(filteredAmpWhaleUnbondingRequests?.map(req => req.asset.amount).reduce((accumulator: number, currentValue: string) => {
```
Comment (0xFable) [inline components/Pages/Dashboard/hooks/useUnbonding.ts:58]: nice bit of map reduce, kudos

File: components/Pages/Dashboard/hooks/useCurrentEpoch.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+      native_token: {
+        denom: string;
+      };
+    };
+  }[];
+}
+interface Epoch {
+  epoch: EpochData;
+}
+export const useCurrentEpoch = (client: Wallet | null) => {
```
Comment (0xFable) [inline components/Pages/Dashboard/hooks/useCurrentEpoch.ts:39]: I like how you be splitting the queries into different hooks. Maybe if we get loads of different types of hooks we could segregate it even more. But this looks great

File: components/Pages/BondingActions/hooks/useTransaction.tsx [hunk 4]
```diff
@@ -0,0 +1,295 @@
...
+        } else {
+          setError('Failed to execute transaction.')
+          message = 'Failed to execute transaction.'
+        }
+
+        toast({
+          title:  (() => {
+            switch (bondingAction) {
+              case ActionType.bond:
+                return "Dashboard Failed.";
```
Comment (0xFable) [inline components/Pages/BondingActions/hooks/useTransaction.tsx:173]: Bonding failed maybe

File: components/Pages/BondingActions/hooks/useTransaction.tsx [hunk 5]
```diff
@@ -0,0 +1,295 @@
...
+        })
+      },
+      onSuccess: (data: any) => {
+        setTxStep(TxStep.Broadcasting)
+        setTxHash(data?.transactionHash)
+        toast({
+          title:  (() => {
+            switch (bondingAction) {
+              case ActionType.bond:
+                return "Dashboard Successful.";
```
Comment (0xFable) [inline components/Pages/BondingActions/hooks/useTransaction.tsx:198]: same?

**Review by 0xFable — APPROVED** [review]
Took quite a while to review the whole thing but think thats because of diff from terra station and new pools being in here. 
Overall looking very good! Most of the contract related stuff like queries and actions looks nicely isolated into their own Hooks. 
Pls add more comments on the lets say more cryptic or longer parts of code. Some of it looks very concise but some explanations will help other devs and the public

File: components/Pages/Dashboard/hooks/getUnbonding.ts [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+export const getUnbonding = async (client: Wallet, address: string, config: Config) => {
+  if (!client || !address) {
+    return null;
+  }
+
+  const unbondingInfos = await fetchUnbonding(client, address, config);
+
+  const unbondingPeriod = 60 * 1_000_000_000;
```
Comment (0xFable) [inline components/Pages/Dashboard/hooks/getUnbonding.ts:36]: make both constants

**Review by 0xFable — CHANGES_REQUESTED** [review]
Changes looking very good ! Pls add some comments and rename a few things for maintainability but otherwise LGTM! Would like to do some internal testing on this.

File: components/Pages/Dashboard/hooks/useWithdrawable.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+export const fetchWithdrawable = async (
+  client: Wallet,
+  address: string,
+): Promise<WithdrawableInfo[]> => {
+
+  const resultAmp: JsonObject = await client.queryContractSmart(BONDING_CONTRACT_ADDRESS, {
+    withdrawable: {address: address, denom: AMP_WHALE_DENOM},
+  });
+  const resultB: JsonObject = await client.queryContractSmart(BONDING_CONTRACT_ADDRESS, {
+    withdrawable: {address: address, denom: B_WHALE_DENOM},
```
[the remark this answers is not in the document]
Comment (worrex) [reply components/Pages/Dashboard/hooks/useWithdrawable.ts]: we do not have these tokens in the testnet and so no corresponding addresses, its just a placeholder

**Review by Sen-Com — APPROVED** [review]

**Pull request merged**